string[] keywords
float32[] similarities
